sim/shaman/items.go

```diff
@@ -131,9 +133,23 @@ var ItemSetVolcanicRegalia = core.NewItemSet(core.ItemSet{
 				})
 			}
 
+			if shaman.ItemSwap.IsEnabled() {
+				shaman.RegisterItemSwapCallback(core.ItemSetSlots,
+					func(sim *core.Simulation, _ proto.ItemSlot) {
+						hasT12Ele2pc := shaman.hasT12Ele2pc()
+
+						if hasT12Ele2pc {
+							aura.Activate(sim)
+						} else {
+							aura.Deactivate(sim)
+						}
+					})
+			}
```

Comment:
It's a bad idea to rely on every single set bonus implementation for every spec adding these handlers manually, will inevitably result in errors down the road. Is there a way to wrap this logic directly into `MakeProcTriggerAura()`, where we add additional `ProcTrigger` inputs to specify a set bonus that the trigger aura is attached to?

Reply:
One way you could implement this is by adding a method like this in sim/core/item_sets.go:
```
func (character *Character) MakeProcTriggerAuraForSetBonus(setName string, numPieces int32, config ProcTrigger) *Aura {
    aura := MakeProcTriggerAura(&character.Unit, config)

    if character.ItemSwap.IsEnabled() {
        character.RegisterItemSwapCallback(ItemSetSlots, func(sim *Simulation, _ proto.ItemSlot) {
            if character.HasActiveSetBonus(setName, numPieces) {
                aura.Activate(sim)
            } else {
                aura.Deactivate(sim)
            }
        })
    }

    return aura
}
```
This could then be substituted anywhere that `core.MakeProcTriggerAura()` is called within x-class/items.go to establish the right pattern for how to define these bonuses.

Reply:
As we've discussed on Discord. You (we) came up for a more solid approach on tackling these :)